way_to_home/custom_user/views.py

```diff
@@ -116,3 +125,53 @@ def signin_google(request):
         return HttpResponse("User was successfully created", status=201)
 
     return HttpResponse("User's data is empty", status=400)
+
+
+@require_http_methods(["POST"])
+def reset_password(request):
+    """Function that provides reset user password"""
+    data = request.body
+    if not update_email_validator(data, 'email'):
+        return False
+    email = data.get('email')
+    user = CustomUser.get_by_email(email=email)
+    if not user:
+        return RESPONSE_400_INVALID_DATA
+    token = create_token(data={'email': user.email})
+    send_email_password_update(user, token)
+    return RESPONSE_200_OK
+
+
+@require_http_methods(['PUT'])
+def confirm_reset_password(request, token):
+    """Function that provides confirm reset user password"""
+    confirm = decode_token(token)
+    if not confirm:
+        return RESPONSE_498_INVALID_TOKEN
+    user = CustomUser.get_by_email(email=confirm.get('email'))
+    if not user:
+        return RESPONSE_400_OBJECT_NOT_FOUND
+
+    data = request.body
+    if not reset_password_validator(data, 'new_password'):
+        return False
+    new_password = data.get('new_password')
+    if not user.check_password(new_password):
+        user.update(password=new_password)
+        send_successful_update_email(user)
+        return RESPONSE_200_OK
+    return RESPONSE_400_INVALID_DATA
+
+
+@require_http_methods(["PUT"])
+def change_password(request):
+    """Function that provides change user password"""
+    user = request.user
+    data = request.body
+    if user.check_password(data["oldPassword"]):
+        if password_validator(data["newPassword"]):
+            user.update(data["newPassword"])
+            user.save()
```

Comment:
update method saves the user, and you do not need to call save method one more time

Reply:
done